boost: Add 1.80.0

jwillikers opened the pull request:
Specify library name and version:  **boost/1.80.0**

Fixes #12564.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

jwillikers commented:
I can't reproduce the `libiconv` issues on Linux and the logs here don't provide enough information for me to diagnose the problem. Unless  someone can spot something wrong with the related patch I had to update, see [here](https://github.com/conan-io/conan-center-index/pull/12807/files#diff-a1f0c325abfbf82c4e7d0df08fca1267cc98d0cf9376362813be62c7e68ff50c), then someone with  macOS will need to provide some support to get this fixed.

TheSalvator commented:
> I can't reproduce the `libiconv` issues on Linux and the logs here don't provide enough information for me to diagnose the problem. Unless someone can spot something wrong with the related patch I had to update, see [here](https://github.com/conan-io/conan-center-index/pull/12807/files#diff-a1f0c325abfbf82c4e7d0df08fca1267cc98d0cf9376362813be62c7e68ff50c), then someone with macOS will need to provide some support to get this fixed.

I have macOS. I think I can test it after 4-5 hours (after work)

TheSalvator commented:
I decided to try to build boost with -d9 (9th level of debug) b2 option.

What a mistake

jwillikers commented:
> I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.

You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.

TheSalvator commented:
> > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> 
> You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.

Here it is
https://github.com/jwillikers/conan-center-index/pull/5

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jwillikers<br/>:x: Ilya Kazakov<br/><hr/>**Ilya Kazakov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan-center-index?pullRequest=12807) it.</sub>

TheSalvator commented:
C'mon. I have already been a contributor. I think i messed up user.name in commit(

jwillikers commented:
> > > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> > 
> > 
> > You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.
> 
> Here it is [jwillikers#5](https://github.com/jwillikers/conan-center-index/pull/5)

@TheSalvator, thanks for figuring out the macOS specific fixes necessary to get this working! I've incorporated your changes in my PR, but you'll need to make sure to sign Conan Center's CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807). Thanks again!

TheSalvator commented:
> > > > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> > > 
> > > 
> > > You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.
> > 
> > 
> > Here it is [jwillikers#5](https://github.com/jwillikers/conan-center-index/pull/5)
> 
> @TheSalvator, thanks for figuring out the macOS specific fixes necessary to get this working! I've incorporated your changes in my PR, but you'll need to make sure to sign Conan Center's CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807). Thanks again!

I messed up again. A already have signed CLA. Can you reset my PR and force push so I change the commit author and reopen the PR

jwillikers commented:
@TheSalvator Done.

TheSalvator commented:
@jwillikers Here you go)
https://github.com/jwillikers/conan-center-index/pull/6

TheSalvator commented:
@jwillikers It's a pleasure working with you. Hope there is no more patches left to submit)

jwillikers commented:
> @jwillikers It's a pleasure working with you. Hope there is no more patches left to submit)

Same!

uilianries commented:
> @uilianries What is `continuous-integration/jenkins/pr-merge`? It takes suspiciously long to complete

@TheSalvator it's the CI build, it will show the final result. It seems be supicious, but Boost takes almost an entire day to be built. One thing is building locally, another thing is generating more than 100 builds for each version.

Both Boost and Qt are our Nemesis in terms of CI, they take both several hours to be built, and for each new commit we need to build all again.

gegles commented:
ANy update on this? Are we sure it's still running?

prince-chrismc commented:
I think we should drop some less popular versions... this way it wont overwhelm CCI so much.
Using my magic goggle to see into the backend, 1.70 and 1.73 are the least popular. I think we should drop those from the build 🔫 at least

https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-for-removing-older-versions-of-a-package

I pinged the CCI team to see if it's stuck, two days is a long time.

prince-chrismc commented on recipes/boost/all/conandata.yml:
```diff
@@ -240,5 +246,33 @@ patches:
+    - patch_file: "patches/1.80.0-0001-filesystem-win-fix-dir-it-net-share.patch"
+      patch_description: "Directory iterators may fail to construct for a network share on Windows prior to 10"
+      patch_type: "official"
+      patch_source: "https://github.com/boostorg/filesystem/issues/245"
```
I absolutely appreciate that someone took the time to add these link ❤️

jwillikers replied:
Thanks! They are super helpful for context and keeping track of what needs to be pulled forward.

puremourning commented:
If it's useful in terms of testing, I have successfully exported this recipe and built it debug and release on both intel and arm Macs, with and without ICU

jgsogo reviewed (approved):
The more people approve, the more resources assigned by the CI 😄  (it would be a nice feature to have)

SpaceIm commented:
Are you sure it's not stuck? Usually boost takes few hours in CI, not days.

SpaceIm commented:
@jgsogo https://github.com/conan-io/conan-center-index/pull/12795#issuecomment-1245161557
